### PR TITLE
warp cube surface to sphere surface

### PR DESCRIPTION
**Pull request by sriharshakandala**
add cube surface to sphere surface warping function. add sphere_mesh to create a sphere surface mesh.

### CODE REVIEW

File: src/Meshes/WarpCubetoSphere.jl [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    b,
+    c,
+    R = max(abs(a), abs(b), abs(c)),
+)
+    r = hypot(a, b, c)
+    x1 = R * a / r
+    x2 = R * b / r
+    x3 = R * c / r
+    return x1, x2, x3
+end
```
Comment (valeriabarra) [inline src/Meshes/WarpCubetoSphere.jl:62]: Can you at least acknowledge authorship of these functions or the repo from which you grabbed them? At least in a comment and/or documentation

Comment (sriharshakandala) [reply src/Meshes/WarpCubetoSphere.jl:62]: @simonbyrne , @valeriabarra : For code obtained from ClimateMachine.jl, should we use the url? e.g.: https://github.com/CliMA/ClimateMachine.jl/blob/master/src/Numerics/Mesh/Topologies.jl

**Review by jakebolewski — CHANGES_REQUESTED** [review]

File: src/Meshes/WarpCubetoSphere.jl [hunk 2]
```diff
@@ -0,0 +1,80 @@
+
+
+"""
+    cubed_sphere_warp(::EquiangularSphereMesh, a, b, c)
+
+Given points `(a, b, c)` on the surface of a cube, warp the points out to a
+spherical shell of radius `R` based on the equiangular gnomonic grid proposed by
+[Ronchi1996]
+
```
[the remark this answers is not in the document]
Comment (jakebolewski) [reply src/Meshes/WarpCubetoSphere.jl:9]: this file should be included in the Meshes module.

File: src/Meshes/CubedSphereMesh.jl [hunk 3]
```diff
@@ -1,3 +1,23 @@
+include("WarpCubetoSphere.jl")
+
+function sphere_mesh(
```
[the remark this answers is not in the document]
Comment (jakebolewski) [reply src/Meshes/CubedSphereMesh.jl:3]: This file should be lower cased and included the Meshes module.

File: src/Meshes/WarpCubetoSphere.jl (hunk 1 above)
Comment (jakebolewski) [reply src/Meshes/WarpCubetoSphere.jl]: All code for both repos is under the same copyright holder and worked on by overlapping authors so we don't need to do much per file / function, but we should create a note in the documentation and README citing the ClimateMachine.jl repo and (eventual) paper: see https://github.com/CliMA/ClimaCore.jl/issues/191.

File: test/grid2d_cs.jl [hunk 4]
```diff
@@ -111,3 +116,33 @@ end
...
+    end
+    @testset "4 elements per edge, equiangular spherical mesh of radius 10; Float type = Float64" begin
+        radius = FT(10)
+        mesh = Mesh2D(EquiangularSphere{FT}(), 4, radius)
+        crad = abs.(sqrt.(sum(mesh.coordinates .^ 2, dims = 2)) .- radius)
+        @test maximum(crad) ≤ 100 * eps(FT)
+    end
+
+    FT = BigFloat
+    @testset "4 elements per edge, equidistant spherical mesh of radius 10; Float type = Float64" begin
```
Comment (jakebolewski) [inline test/grid2d_cs.jl:136]: Float type: BigFloat

File: src/Meshes/Meshes.jl [hunk 5]
```diff
@@ -21,6 +27,10 @@ in a file, it would contain the filename.
 """
 abstract type AbstractMesh{FT} end
 
+abstract type ASphere{FT} end
```
Comment (jakebolewski) [inline src/Meshes/Meshes.jl:30]: `ASphere` -> `AbstractSphere{FT}`

**Review by jakebolewski — APPROVED** [review]

**Pull request merged**